src/main/java/org/betonquest/betonquest/item/QuestItem.java

```diff
@@ -417,8 +445,30 @@ public boolean compare(final ItemStack item) {
         }
         if (meta instanceof SkullMeta) {
             final SkullMeta skullMeta = (SkullMeta) item.getItemMeta();
-            if (!head.check(skullMeta.getOwner())) {
-                return false;
+            if (!head.checkOwner(skullMeta.getOwner())) {
+                val ownerProfile = skullMeta.getOwningPlayer();
+                val playerProfile = skullMeta.getPlayerProfile();
+                if (ownerProfile != null) {
+                    // For Bukkit / Spigot Server
```

Comment:
I don't see a way to support the textures URL through the spigot API.

Reply:
That's ok but then we need two variants of the HeadHandler, one for Spigot and one for Paper.
The `Template Method` pattern could be used here. A different implementation could be loaded based on if `PaperLib.isPaper()` is true.

Reply:
take a look at the latest changes